src/components/Autocomplete/components/ComboBox/ComboBox.tsx

```diff
@@ -53,467 +42,337 @@ export interface ComboBoxProps {
   /** Callback when the end of the list is reached */
   onEndReached?(): void;
 }
+export function ComboBox({
+  id: idProp,
+  options,
+  selected,
+  textField,
+  preferredPosition,
+  listTitle,
+  allowMultiple,
+  actionsBefore,
+  actionsAfter,
+  contentBefore,
+  contentAfter,
+  emptyState,
+  onSelect,
+  onEndReached,
+}: ComboBoxProps) {
+  const [selectedIndex, setSelectedIndex] = useState(-1);
+  const [selectedOptions, setSelectedOptions] = useState(selected);
+  const [navigableOptions, setNavigableOptions] = useState<
+    (OptionDescriptor | ActionListItemDescriptor)[]
+  >([]);
+  const {
+    value: popoverActive,
+    setTrue: forcePopoverActiveTrue,
+    setFalse: forcePopoverActiveFalse,
+  } = useToggle(false);
+
+  const id = useUniqueId('ComboBox', idProp);
+
+  const getActionsWithIds = useCallback(
+    (
+      actions: ActionListItemDescriptor[],
+      before?: boolean,
+    ): ActionListItemDescriptor[] => {
```

Comment:
TS can infer the return type fine, no need to be explicit

```suggestion
    ) => {
```